### PR TITLE
chore: add a unit test for conditional and string equals projections

### PR DESCRIPTION
**Pull request by tim-mwangi**
Tests specifically for returning "null" string.

### CODE REVIEW

File: hypertrace-trace-enricher/trace-reader/src/test/java/org/hypertrace/trace/reader/attributes/DefaultValueResolverTest.java [hunk 1]
```diff
@@ -176,6 +179,77 @@ void resolvesExpressionProjections() {
...
+                                                .setOperator(ProjectionOperator.PROJECTION_OPERATOR_STRING_EQUALS)
+                                                .addArguments(Projection.newBuilder().setAttributeId("TEST_SCOPE.second"))
+                                                .addArguments(Projection.newBuilder().setLiteral(LiteralValue.newBuilder().setStringValue("foo")))
+                                        )
+                                    )
+                                    .addArguments(
+                                        Projection.newBuilder().setAttributeId("TEST_SCOPE.first"))
+                                    .addArguments(
+                                        Projection.newBuilder()
+                                            .setLiteral(LiteralValue.newBuilder().setStringValue("null")))
```
Comment (tim-mwangi) [inline hypertrace-trace-enricher/trace-reader/src/test/java/org/hypertrace/trace/reader/attributes/DefaultValueResolverTest.java:205]: @aaron-steinfeld is this what you meant?

Comment (aaron-steinfeld) [reply hypertrace-trace-enricher/trace-reader/src/test/java/org/hypertrace/trace/reader/attributes/DefaultValueResolverTest.java:205]: Yes, but this is verifying the behavior I was afraid of. If I try to use the attribute reader to get a value and it returns a string literal of "null", the burden is on every consumer to differentiate that from a valid string. For example, in the entity reader, we'd try to create an entity with the id "null" if we don't look for this special string, which is based on pinot's conception of null.

The real behavior we want IMO is one of these:
- For the resolved single to be an error (the current api supports this, but it means the actual projection would need to throw which it doesn't currently)
- For the resolved single to be an unset literal value or a currently non-existent null-typed literal
- Change the attribute reader api to return a Maybe (this is a slight tweak of the first variation above, still requires the actual projection code to indicate an invalid result)

**Comment by tim-mwangi** [comment]
@aaron-steinfeld is this still necessary?

**Comment by aaron-steinfeld** [comment]
> @aaron-steinfeld is this still necessary?

I don't think so - I added general tests for the null case in both the attribute and ingester relevant areas.

**Comment by tim-mwangi** [comment]
> > @aaron-steinfeld is this still necessary?
> 
> I don't think so - I added general tests for the null case in both the attribute and ingester relevant areas.

Ok. Will close it out.

**Pull request closed without merging**